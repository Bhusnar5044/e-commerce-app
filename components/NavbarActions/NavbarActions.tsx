'use client';

import { ShoppingBag } from 'lucide-react';
import { useRouter } from 'next/navigation';
import { memo, useCallback, useEffect, useState } from 'react';

import Button from '@/components/ui/Button';
import { useAppSelector } from '@/redux';

export const NavbarActions = memo(() => {
    const [isMounted, setIsMounted] = useState(false);
    const cart = useAppSelector((state) => state.cart);
    const router = useRouter();

    const handleNavigation = useCallback(() => {
        router.push('/cart');
    }, [router]);

    useEffect(() => {
        setIsMounted(true);
    }, []);

    if (!isMounted) {
        return null;
    }

    return (
        <div className="ml-auto flex items-center gap-x-4">
            <Button onClick={handleNavigation} className="flex items-center rounded-full bg-black px-4 py-2">
                <ShoppingBag size={20} color="white" />
                <span className="ml-2 text-sm font-medium text-white">{cart.items.length}</span>
            </Button>
        </div>
    );
});

NavbarActions.displayName = 'NavbarActions';
