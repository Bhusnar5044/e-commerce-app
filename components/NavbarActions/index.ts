export { NavbarActions as default } from './NavbarActions';
