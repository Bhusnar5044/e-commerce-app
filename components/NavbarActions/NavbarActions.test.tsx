import { renderWithStore } from '@test-utils';
import { NavbarActions } from './NavbarActions';

jest.mock('next/navigation', () => ({
    ...require('next-router-mock'),
}));
describe('NavbarActions', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<NavbarActions />);
        expect(container).toMatchSnapshot();
    });
});
