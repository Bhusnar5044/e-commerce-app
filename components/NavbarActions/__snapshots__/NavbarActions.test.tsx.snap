// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarActions should render properly 1`] = `
<div>
  <div
    class="ml-auto flex items-center gap-x-4"
  >
    <button
      class="w-auto border border-transparent disabled:cursor-not-allowed disabled:opacity-50 text-white font-semibold hover:opacity-75 transition flex items-center rounded-full bg-black px-4 py-2"
      type="button"
    >
      <svg
        class="lucide lucide-shopping-bag"
        fill="none"
        height="20"
        stroke="white"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
        />
        <path
          d="M3 6h18"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
        />
      </svg>
      <span
        class="ml-2 text-sm font-medium text-white"
      >
        0
      </span>
    </button>
  </div>
</div>
`;
